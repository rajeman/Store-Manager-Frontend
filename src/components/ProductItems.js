import React from 'react';
import { connect } from 'react-redux';
import Product from './Product'
import { fetchProducts } from '../actions/products';

class ProductItems extends React.Component {
    componentDidMount() {
        this.props.dispatch(fetchProducts())
    }
    render() {
        const { products } = this.props.products;
        return (
            <div className="container">
                <div className="wrapper">
                    <div className="items-box">
                        {products.map((product) => {
                            return (
                                <Product key={product.product_id} product = {product}/>
                            );
                        })}
                    </div>
                </div>
            </div>
        );
    }
}

const mapStateToProps = (state) => state;
export default connect(mapStateToProps)(ProductItems);
