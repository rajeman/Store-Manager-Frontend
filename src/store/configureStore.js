import { createStore, combineReducers, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';
import productReducer from '../reducers/products';
import authReducer from '../reducers/auth';
import saleReducer from '../reducers/records';
import navigationReducer from '../reducers/navigation';
import attendantReducer from '../reducers/attendants';
import cartReducer from '../reducers/cart';

export default () => createStore(
  combineReducers({
    products: productReducer,
    auth: authReducer,
    navigation: navigationReducer,
    sales: saleReducer,
    attendants: attendantReducer,
    cart: cartReducer,
  }), applyMiddleware(thunk),
);
