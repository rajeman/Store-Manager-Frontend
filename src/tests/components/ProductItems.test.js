import React from 'react';
import { shallow, mount } from 'enzyme';
import { Provider } from 'react-redux';
import thunk from 'redux-thunk';
import configureStore from 'redux-mock-store';
import axios from 'axios';
import ProductItems from '../../components/ProductItems';

jest.mock('axios');

describe('PRODUCT TEST SUITE', () => {
  it('should render the product items when component mounts', () => {
    const response = {
      data:
      {
        message: 'successfully fetched user info',
      }
    };
    axios.get.mockResolvedValue(response);
    const initialState = {
      products: {
        products: [{
          product_id: 7
        }]
      }
    };
    const mockStore = configureStore([thunk]);
    const store = mockStore(initialState);
    const component = mount(
      <Provider store={store}>
        <ProductItems />
      </Provider>
    );
    expect(component).toMatchSnapshot();
  });

  it('should navigate to the single product page when a product is selected', () => {
    const response = {
      data:
      {
        message: 'successfully fetched user info',
      }
    };
    axios.get.mockResolvedValue(response);
    const initialState = {
      products: {
        products: [{
          product_id: 7
        }]
      }
    };
    const mockStore = configureStore([thunk]);
    const store = mockStore(initialState);
    const component = mount(
      <Provider store={store}>
        <ProductItems />
      </Provider>
    );
    component.find('div[className="item"]').simulate('click');
  });
});
