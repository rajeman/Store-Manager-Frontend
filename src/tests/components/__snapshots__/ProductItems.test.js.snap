// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PRODUCT TEST SUITE should render the product items when component mounts 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(ProductItems)>
    <ProductItems
      dispatch={[Function]}
      products={
        Object {
          "products": Array [
            Object {
              "product_id": 7,
            },
          ],
        }
      }
    >
      <div
        className="container"
      >
        <div
          className="wrapper"
        >
          <div
            className="items-box"
          >
            <Component
              key="7"
              product={
                Object {
                  "product_id": 7,
                }
              }
            >
              <div
                className="item"
                onClick={[Function]}
              >
                <img
                  src={Object {}}
                />
                <div
                  className="quant-avail"
                >
                   
                  available
                </div>
                <div
                  className="name-description"
                />
                <div
                  className="price"
                >
                  $
                  <span
                    className="actual-price"
                  />
                   
                  per item
                </div>
              </div>
            </Component>
          </div>
        </div>
      </div>
    </ProductItems>
  </Connect(ProductItems)>
</Provider>
`;
